### PR TITLE
feat(connector): [Novalnet] Integrate wallets Paypal and Googlepay

### PR DESCRIPTION
**Pull request by cookieg13**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
1. Integrated Googlepay and Paypal for Novalnet 
2. Integrated mandates for Googlepay and Paypal

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
https://github.com/juspay/hyperswitch-cloud/issues/6731

## How did you test it?

Tested manually.
Test cases:
1. Googlepay payment
    - manual
    - automatic
    - mandates
2. Paypal 
    - manual
    - automatic
    - mandates

[Manual Testing Doc](https://docs.google.com/document/d/1E1nkLfc8qg0htkxOkwbsA0f9B5QBoECxJ4swH1WElgw/edit?usp=sharing)

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs [hunk 1]
```diff
@@ -402,12 +565,20 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
...
                     .and_then(|data| data.tid.map(|tid| tid.expose().to_string()));
+
+                let mandate_reference_id = NovalnetPaymentsResponseTransactionData::get_token(
+                    item.response.transaction.clone().as_ref(),
+                );
+
                 let transaction_status = item
                     .response
                     .transaction
+                    .clone()
```
Comment (srujanchikke) [inline crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs:576]: can we pass reference here ?

File: crates/hyperswitch_connectors/src/connectors/novalnet.rs [hunk 2]
```diff
@@ -195,8 +195,11 @@ impl ConnectorValidation for Novalnet {
         pm_type: Option<enums::PaymentMethodType>,
         pm_data: PaymentMethodData,
     ) -> CustomResult<(), errors::ConnectorError> {
-        let mandate_supported_pmd: HashSet<PaymentMethodDataType> =
-            HashSet::from([PaymentMethodDataType::Card]);
+        let mandate_supported_pmd: HashSet<PaymentMethodDataType> = HashSet::from([
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/connectors/novalnet.rs:198]: Please add the changes in `crates/connector_configs/toml` as well

File: crates/hyperswitch_connectors/src/connectors/novalnet.rs [hunk 3]
```diff
@@ -815,8 +818,15 @@ impl webhooks::IncomingWebhook for Novalnet {
...
+            let parent_tid =
+                notif
+                    .event
+                    .parent_tid
+                    .ok_or(errors::ConnectorError::MissingRequiredField {
+                        field_name: "parent_tid",
+                    })?;
             Ok(api_models::webhooks::ObjectReferenceId::RefundId(
-                api_models::webhooks::RefundIdType::ConnectorRefundId(notif.event.tid.to_string()),
+                api_models::webhooks::RefundIdType::ConnectorRefundId(parent_tid.to_string()),
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/connectors/novalnet.rs:829]: Are we supposed to populate `ConnectorRefundId` with `parent_tid`?

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Comment by deepanshu-iiitu** [comment]
@cookieg13 
Did you test Google Pay and Paypal Flows?

**Review by srujanchikke — DISMISSED** [review]

**Comment by cookieg13** [comment]
> @cookieg13 Did you test Google Pay and Paypal Flows?

yes tested locally, webhooks and mandates are tested too

File: crates/hyperswitch_connectors/src/connectors/novalnet.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (cookieg13) [reply crates/hyperswitch_connectors/src/connectors/novalnet.rs]: according to docs,
[event.tid](https://developer.novalnet.com/asynchronousnotification/parameterspassed#response-event-tid) - The TID which is relevant to this event.
[event.parent_tid](https://developer.novalnet.com/asynchronousnotification/parameterspassed#response-event-parent_tid) - During the course of followup events like chargebacks, credit entry etc.., this field will contain the value of the original/initial transaction TID.

I have removed this change. Not needed.

**Review by srujanchikke — DISMISSED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]

**Comment by Gnanasundari24** [comment]
@cookieg13 You add the response aswell then only will get to know whether its worked or not

**Comment by cookieg13** [comment]
> @cookieg13 Can you add the response aswell then only will get to know whether its worked or not. and add the mandate req and res also

added testing doc with req and responses in readme

**Review by srujanchikke — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**